src/libraries/System.Text.Json/src/System/Text/Json/AppContextSwitchHelper.cs

```diff
@@ -16,5 +16,11 @@ internal static class AppContextSwitchHelper
                 switchName: "System.Text.Json.Serialization.RespectNullableAnnotationsDefault",
                 isEnabled: out bool value)
             ? value : false;
+
+        public static bool RespectRequiredConstructorParametersDefault { get; } =
+            AppContext.TryGetSwitch(
+                switchName: "System.Text.Json.Serialization.RespectRequiredConstructorParametersDefault",
```

Comment:
that seems like it should be on the serializer options in the long term?

Reply:
oh, I see this is the default for the Default options?

Reply:
Yep yep